test/plots/date-intervals.ts

```diff
@@ -0,0 +1,26 @@
+import * as Plot from "@observablehq/plot";
+import * as d3 from "d3";
+
+export async function intervalWeekMonth() {
+  const data = await d3.csv("data/bls-industry-unemployment.csv");
+  return Plot.plot({
+    marginLeft: 80,
+    color: {type: "linear", scheme: "blues"},
+    marks: [
+      {label: "month", interval: "month" as Plot.TimeIntervalName},
+      {label: "d3.utcMonth", interval: d3.utcMonth as Plot.IntervalImplementation<Date>},
+      {label: "week", interval: "week" as Plot.TimeIntervalName},
+      {label: "d3.utcWeek", interval: d3.utcWeek as Plot.IntervalImplementation<Date>}
```

Comment:
the way I handle types here is probably wrong

Reply:
You should use a const assertion instead, like so:

```js
    marks: (
      [
        {label: "month", interval: "month"},
        {label: "d3.utcMonth", interval: d3.utcMonth},
        {label: "week", interval: "week"},
        {label: "d3.utcWeek", interval: d3.utcWeek}
      ] as const
    ).map(({label, interval}) =>
      Plot.barX(data, {
        x: "date",
        filter: (d) => d.industry === "Construction",
        interval,
        fill: "unemployed",
        title: "unemployed",
        inset: 0,
        fy: () => label
      })
    )
```